### PR TITLE
Quilt3 catalog url

### PR DESCRIPTION
**Pull request by kevinemoore**
## Description
Adds an argument to `quilt3 catalog` to specify an S3 URL to browse to in the local catalog.

### CODE REVIEW

File: api/python/quilt3/util.py [hunk 1]
```diff
@@ -433,3 +433,11 @@ def validate_key(key):
                 f"Invalid key {key!r}. "
                 f"A package entry key cannot contain a file or folder named '.' or '..' in its path."
             )
+
+def catalog_s3_url(s3_url):
+    bucket, path, version_id = parse_s3_url(urlparse(s3_url))
+    catalog_url = f"/b/{bucket}/tree/{path}"
```
Comment (dimaryaz) [inline api/python/quilt3/util.py]: Need to `quote` the params - `bucket` and `path`

File: api/python/quilt3/main.py [hunk 2]
```diff
@@ -3,15 +3,15 @@
 """
 
 import argparse
-import dns.resolver
 import subprocess
 import sys
 
+import dns.resolver
```
Comment (dimaryaz) [inline api/python/quilt3/main.py]: Why did you move it? It's a built-it package, right?

File: api/python/quilt3/util.py [hunk 3]
```diff
@@ -433,3 +433,11 @@ def validate_key(key):
                 f"Invalid key {key!r}. "
                 f"A package entry key cannot contain a file or folder named '.' or '..' in its path."
             )
+
+def catalog_s3_url(s3_url):
```
Comment (dimaryaz) [inline api/python/quilt3/util.py]: Just a thought - but might be cleaner to pass in the catalog URL, and have this function return the full URL, not just the path and query params.

File: api/python/quilt3/main.py (hunk 2 above)
Comment (kevinemoore) [reply api/python/quilt3/main.py]: No, I think it's an extra lib, though one we already depended on.

Comment (armandmcqueen) [reply api/python/quilt3/main.py]: We need to add it to `setup.py`
